### PR TITLE
Add copy-to-clipboard documentation

### PR DESCRIPTION
**Pull request by Blackbaud-TimPepper**

### CODE REVIEW

**Comment by Blackbaud-BrandonJones** [comment]
Completely non blocking but, what are your thoughts on adding a text area to paste to for the demo?
screen shot 2018-07-24 at 9 31 07 am

File: src/app/content/clipboard/index.html [hunk 1]
```diff
@@ -0,0 +1,63 @@
+<stache
+  pageTitle="Copy to Clipboard"
+  layout="sidebar"
+  showBreadcrumbs="true">
+
+  <stache-page-summary>
+    <p>
+      The <stache-code>copy-to-clipboard</stache-code> component allows users to copy content to their clipboard. The content to be copied must be wrapped in an element that can be referenced with the <stache-code>#identifier</stache-code> notation. A <stache-code>stache-copy-to-clipboard</stache-code> element must then be added and contain a <stache-code>[copyTarget]</stache-code> attribute that references the identifier from the previously created element. You can optionally include a <stache-code>buttonText</stache-code> and <stache-code>buttonClickedText</stache-code> attribute to alter the button text. When the button is clicked, all text content within the <stache-code>copyTarget</stache-code> element is copied to the user's clipboard with formatting intact. The content can then be pasted in to any field that can receive text.
```
Comment (Blackbaud-BrandonJones) [inline src/app/content/clipboard/index.html]: Should we make a note here that the only elements that can receive that `#identifier` are HTML elements?  And if you want to copy the content of a component you can wrap the component in an HTML element such as a DIV and add the identifier to that?

File: src/app/content/clipboard/index.html [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    </p>
+  </stache-page-summary>
+
+  <h3>
+    Copy to clipboard properties
```
Comment (blackbaud-johnly) [inline src/app/content/clipboard/index.html]: Based on how we handle this elsewhere, "Copy-to-clipboard" should be hyphenated here.

File: src/app/content/clipboard/index.html (hunk 1 above)
Comment (Blackbaud-DeniseCisneros) [reply src/app/content/clipboard/index.html]: I like suggestion of providing an example, especially because it matches the format of the rest of the docs where it's explanation, sample, code.

As for the `#identifier` note, I think that's worth calling out especially if ti's something that could potentially trip someone up/make things not work for them, or if it's a scenario that's very likely to occur.

File: src/app/content/clipboard/index.html [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+  </h3>
+
+  <ul>
+    <li>
+      <stache-code>copyTarget</stache-code> — Section of content to be copied.
```
Comment (blackbaud-johnly) [inline src/app/content/clipboard/index.html]: This should follow the pattern of a sentence fragment that starts with a verb, and I'd change "to be copied" to "to copy" to use active voice, and I don't think we really need "section of": "Specifies the content to copy."

Also, I think we should add a second sentence to briefly explain that this property needs to square brackets. (Hopefully something generic that we can use elsewhere too.) What do you think of: "This property binds data, so you must use brackets to evaluate the template expression: `[copyTarget]`. Without the brackets, Angular treats the string as a constant and initializes the target property with that string instead of evaluating the string."

File: src/app/content/clipboard/index.html [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    </li>
+    <li>
+      <stache-code>buttonText</stache-code> — <i>(Optional)</i> Text to be displayed by copy-to-clipboard button.
```
Comment (blackbaud-johnly) [inline src/app/content/clipboard/index.html]: Should start with a verb, and I'd reword to avoid passive voice: "Specifies the initial label for the copy-to-clipboard button."

File: src/app/content/clipboard/index.html [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+    </li>
+  </ul>
+
+  <stache-page-anchor>
+    Demo
+  </stache-page-anchor>
+
+  <div class="stache-demo-page-result">
+    <div #divReference>
+      Copy text within a div
```
Comment (blackbaud-johnly) [inline src/app/content/clipboard/index.html]: I'd change this to "Text within a div element to copy to the clipboard"

File: src/app/content/clipboard/index.html [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+      <p>
+        Nested Content within the div
```
Comment (blackbaud-johnly) [inline src/app/content/clipboard/index.html]: I'd change this to "Nested text within a div element to copy to the clipboard"

File: src/app/content/clipboard/index.html [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+      </p>
+    </div>
+    <stache-copy-to-clipboard buttonText="Copy From Div" buttonClickedText="Copied!" [copyTarget]="divReference"></stache-copy-to-clipboard>
```
Comment (blackbaud-johnly) [inline src/app/content/clipboard/index.html]: I'd change the initial button label to "Copy to clipboard" ... If not, then use sentence-style capitalization: "Copy from div"

File: src/app/content/clipboard/index.html [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+  </stache-page-anchor>
+
+  <div class="stache-demo-code">
+    <sky-tabset>
+      <sky-tab
+        tabHeading="sample">
+        <stache-code-block
+          languageType="markup">
+          &lt;div #divReference&gt;
+            Copy text within a div
```
Comment (blackbaud-johnly) [inline src/app/content/clipboard/index.html]: Note: make same changes to the sample as recommended above in the demo.

File: src/app/content/clipboard/index.html [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+
+  <ul>
+    <li>
+      <stache-code>copyTarget</stache-code> — Section of content to be copied.
+    </li>
+    <li>
+      <stache-code>buttonText</stache-code> — <i>(Optional)</i> Text to be displayed by copy-to-clipboard button.
+    </li>
+    <li>
+      <stache-code>buttonClickedText</stache-code> — <i>(Optional)</i> Text to be displayed by copy-to-clipboard button once it has been clicked.
```
Comment (blackbaud-johnly) [inline src/app/content/clipboard/index.html]: Should start with a verb, and I'd reword to avoid passive voice: "Specifies a label to display on the copy-to-clipboard button after users click it."

File: src/app/content/clipboard/index.html (hunk 1 above)
Comment (blackbaud-johnly) [inline src/app/content/clipboard/index.html]: I started to provide a list of edits for this paragraph, but I decided it would probably be easier just to offer an alternate paragraph. I've listed some of the edits below in case you want to see the rationale for the edits, but here's the alternate version: "The copy-to-clipboard component creates a button that allows users to copy content to the clipboard. To implement the component, you wrap text in an element and specify an `#identifier` for that element. Then you place a `stache-copy-to-clipboard` element on the page and specify the content to copy by referencing its identifier in the `copyTarget` attribute. When users click the copy-to-clipboard button, the component copies the text to the user's clipboard with formatting intact. If necessary, you can specify different labels for before and after users click the button."

In the first sentence: The first code snippet should not be a code snippet because it's a generic reference and it's missing the "stache-". I'd also change "their" to "the." (If not, then "clipboard" should be "clipboards.") ... In the second sentence, I'd flip the sentence around to use active voice? ... I'd tweak the third sentence to use active voice, tighten up, and make it clearer that this is a continuation of the action in the previous sentence. ... I'd move the fourth sentence to the end and tighten it up to be more generic but also explain more what the options do. ... I'd rewrite the last sentence to use active voice adn to tighten up.

File: src/app/content/clipboard/index.html (hunk 3 above)
Comment (Blackbaud-TimPepper) [reply src/app/content/clipboard/index.html]: Maybe this, for the end: ```Without the brackets, Angular treats the value as a string and initializes the target property with that string instead of evaluating the data value.```

I think to a developer a "string" is already a constant, where as what we are actually passing in here is a variable with dynamic data.

Comment (blackbaud-johnly) [reply src/app/content/clipboard/index.html]: I like it! ... If you think that sentence plus the preceding sentence works here, then I can go back and add that info to other properties that require brackets.

File: src/app/content/clipboard/index.html [hunk 10]
```diff
@@ -0,0 +1,63 @@
+<stache
+  pageTitle="Copy to Clipboard"
```
[the remark this answers is not in the document]
Comment (blackbaud-johnly) [reply src/app/content/clipboard/index.html]: I made an update to address this.

File: src/app/content/clipboard/index.html [hunk 11]
```diff
@@ -0,0 +1,77 @@
...
+    <li>
+      <stache-code>copyTarget</stache-code> — Specifies content to copy.
+      <p>
+        <sky-alert alertType="info">
+          This property binds data, so you must use brackets to evaluate the template expression: <stache-code>[copyTarget]</stache-code>. Without the brackets, Angular treats the value as a string and initializes the target property with the string value instead of evaluating the data value.
+        </sky-alert>
+      </p>
+    </li>
+    <li>
+      <stache-code>buttonText</stache-code> — <i>(Optional)</i> Specifies text to be displayed by copy-to-clipboard button.
```
Comment (blackbaud-johnly) [inline src/app/content/clipboard/index.html]: This strikes me as slightly unclear. It reads like it could be text that is displayed after you click the button. I think I suggested something like "Specifies a label for the copy-to-clipboard button." If "label" doesn't work, I'd suggest changing this to "Specifies text to display on the copy-to-clipboard button."

File: src/app/content/clipboard/index.html [hunk 12]
```diff
@@ -0,0 +1,77 @@
...
+    </li>
+    <li>
+      <stache-code>buttonClickedText</stache-code> — <i>(Optional)</i> Specifies text to be displayed by copy-to-clipboard button once it has been clicked.
```
Comment (blackbaud-johnly) [inline src/app/content/clipboard/index.html]: Same note as above. If "label" doesn't work, I'd suggest changing this to "Specifies text to display on the copy-to-clipboard button after users click it."
